src/main/scala/cs332/distributedsorting/master/Master.scala

```diff
@@ -67,36 +69,39 @@ class Master(executionContext: ExecutionContext, val numClient: Int) { self =>
   }
 
 
-  private def addData(data : Array[Byte],ipAddress : String) : Boolean = {
+  private def addData(data_ : Array[Byte],ipAddress : String) : Boolean = {
     this.synchronized({
-      if (slaves.filter(x=>x.ip == ipAddress).isEmpty)
-        System.out.println("we receive data from a client we did not register")
-        return false 
-      count +=1
-      this.data :+ data.grouped(10).toList
+      /*if (this.slaves.toList.filter(x =>x.ip == ipAddress).isEmpty)
+      Master.logger.info("we receive data from a client we did not register")
+        return false*/
+      this.count +=1
+      this.data = this.data ++ data_.grouped(10).toList
+      System.out.println(this.data.length)
+      Master.logger.info("count is : " + count.toString())
       if(count == numClient)
         System.out.println("we receive all the data")
       return true
     })
   }
 
   private def createPartition(): Map[String, (Array[Byte], Array[Byte])] = {
+    assert(this.count == this.numClient)
     val mindata : Array[Byte] = Array(0,0,0,0,0,0,0,0,0,0)
     val maxdata : Array[Byte] = Array(-2^7,-2^7,-2^7,-2^7,-2^7,-2^7,-2^7,-2^7,-2^7,-2^7)
     // we first need to sort the data list
     this.data.sorted(KeyOrdering)
     // then we can create the partiton
-    val range:Int =(data.length/count) 
+    val range:Int =(data.length/this.count) 
     var loop = 0
-    for (slave <- slaves){
+    for (slave <- this.slaves.toList){
       if (loop == 0){
-        this.partition.put(slave.toString(), (mindata, data((loop+1)*range)))
+        this.partition.put(slave.toString(), (mindata, this.data((loop+1)*range -1)))
       }
-      if (loop == count - 1){
-        this.partition.put(slave.toString(), (data(loop*range), maxdata))
+      else if (loop == count-1){
+        this.partition.put(slave.toString(), (data((loop)*range -1), maxdata))
       }
       else{
-        this.partition.put(slave.toString(), (data(loop*range), data((loop+1)*range)))
+        this.partition.put(slave.toString(), (data(loop*range -1), data((loop+1)*range -1)))
       } 
```

Comment:
Why did you add -1 for every lowerbound and upperbound?
Is the lowerbound and upperbound is included or excluded? (ex. lowerbound: 3, upperbound: 7. It means 3 <= x <= 7 if all bounds mean included.)

Reply:
lower bound and upperbound are included that's why I add -1 to upperbound to not overlapp with the lowerbound of the next partition

Reply:
But there is -1 in the lowerbound variable either. Is it intended?

Reply:
yes my mistake I changed it